source/extensions/filters/listener/dynamic_modules/factory.cc

```diff
@@ -0,0 +1,66 @@
+#include "source/extensions/filters/listener/dynamic_modules/factory.h"
+
+#include "envoy/registry/registry.h"
+
+#include "source/common/protobuf/utility.h"
+#include "source/extensions/filters/listener/dynamic_modules/filter.h"
+#include "source/extensions/filters/listener/dynamic_modules/filter_config.h"
+
+namespace Envoy {
+namespace Server {
+namespace Configuration {
+
+Network::ListenerFilterFactoryCb
+DynamicModuleListenerFilterConfigFactory::createListenerFilterFactoryFromProto(
+    const Protobuf::Message& message,
+    const Network::ListenerFilterMatcherSharedPtr& listener_filter_matcher,
+    ListenerFactoryContext& context) {
+
+  const auto& proto_config = MessageUtil::downcastAndValidate<const ListenerFilterConfig&>(
+      message, context.messageValidationVisitor());
+
+  const auto& module_config = proto_config.dynamic_module_config();
+  auto dynamic_module = Extensions::DynamicModules::newDynamicModuleByName(
+      module_config.name(), module_config.do_not_close(), module_config.load_globally());
+  if (!dynamic_module.ok()) {
+    throw EnvoyException("Failed to load dynamic module: " +
```

Comment:
oh - don't we have a StatusOr variant factory for listener filter?

Reply:
Yeah, it doesn't exist yet.